packages/components/organisms/f-checkout/src/components/ErrorDialog.vue

```diff
@@ -1,58 +1,106 @@
 <template>
-    <mega-modal
-        :is-open="isOpen"
-        data-test-id="checkout-issue-modal"
-        has-overlay
-        v-bind="$attrs"
-        v-on="$listeners">
-        <h3
-            data-test-id="checkout-issue-modal-title"
-            class="u-noSpacing">
-            {{ $t(`errorMessages.checkoutIssues.${errorCode}.title`) }}
-        </h3>
-
-        <p data-test-id="checkout-issue-modal-message">
-            {{ $t(`errorMessages.checkoutIssues.${errorCode}.message`) }}
-        </p>
-
-        <f-button
-            :class="$style['c-checkout-redirectButton']"
-            button-size="large"
-            data-test-id="redirect-to-menu-button"
-            @click.native="handleButtonClick">
-            {{ $t(`errorMessages.checkoutIssues.${errorCode}.buttonText`) }}
-        </f-button>
-    </mega-modal>
+    <div>
+        <mega-modal
+            data-test-id="checkout-issue-modal"
+            has-overlay
+            :is-open="shouldShowModal"
+            @close="handleErrorDialogButtonClick"
+        >
+            <h3
+                data-test-id="checkout-issue-modal-title"
+                class="u-noSpacing"
+            >
+                {{ $t(`errorMessages.checkoutIssues.${errorCode}.title`) }}
+            </h3>
+
+            <p data-test-id="checkout-issue-modal-message">
+                {{ $t(`errorMessages.checkoutIssues.${errorCode}.message`) }}
+            </p>
+
+            <f-button
+                :class="$style['c-checkout-redirectButton']"
+                button-size="large"
+                data-test-id="redirect-to-menu-button"
+                @click.native="handleErrorDialogButtonClick"
+            >
+                {{ $t(`errorMessages.checkoutIssues.${errorCode}.buttonText`) }}
+            </f-button>
+        </mega-modal>
+
+        <alert
+            v-if="shouldShowAlert"
+            type="danger"
+            :class="$style['c-checkout-alert']"
+            :heading="$t('errorMessages.errorHeading')"
+        >
+            {{ errorCode }}
+        </alert>
+    </div>
 </template>
 
 <script>
+import Alert from '@justeat/f-alert';
+import '@justeat/f-alert/dist/f-alert.css';
 import MegaModal from '@justeat/f-mega-modal';
 import '@justeat/f-mega-modal/dist/f-mega-modal.css';
 import FButton from '@justeat/f-button';
 import '@justeat/f-button/dist/f-button.css';
+import { mapActions, mapState } from 'vuex';
+import {
+    VUEX_CHECKOUT_MODULE
+} from '../constants';
 
 import EventNames from '../event-names';
 
 export default {
     components: {
+        Alert,
         MegaModal,
         FButton
     },
 
-    props: {
-        isOpen: {
-            type: Boolean,
-            default: false
+    computed: {
+        ...mapState(VUEX_CHECKOUT_MODULE, [
+            'displayError',
```

Comment:
If this is a boolean value, then let's name it `shouldDisplayError`.

Reply:
This one was the actual message. I'll come up with a better name so that it makes more sense. Thanks!

Reply:
changed to `message`